src/rules/no_nested_string_interpolation.coffee

```diff
@@ -17,27 +17,38 @@ module.exports = class NoNestedStringInterpolation
             </pre>
             '''
 
-    tokens: ['STRING_START', 'STRING_END']
+    tokens: ['CSX_TAG', 'CALL_START', 'CALL_END', 'STRING_START', 'STRING_END']
 
     constructor: ->
-        @startedStrings = 0
-        @generatedError = false
-
-    lintToken: ([type], tokenApi) ->
-        if type is 'STRING_START'
-            @trackStringStart()
+        @blocks = []
+
+    lintToken: (token, tokenApi) ->
```

Comment:
I rewrote this function from scratch. To test for nested interpolation the old way is problematic since {} code blocks inside CSX tags are treated as nested interpolation in the old scheme. 
In the new way, we keep track of visited string_start and string_end per csx tag. Each time we encounter a csx tag, we nest to another 'block', with its own tracking.